Tutorial (#293)
cyclus/cyclus.github.com: source/user/tutorial/add_proto.rst

+-------------------------+---------------------------+
| ``feed_commod``         | ``u-ore``                 |
+-------------------------+---------------------------+
| ``feed_recipe``         | ``nat-u``                 |
+-------------------------+---------------------------+
| ``product_commod``      | ``fresh-uox``             |
+-------------------------+---------------------------+
| ``tails_commod``        | ``tails``                 |
+-------------------------+---------------------------+
| ``max_feed_inventory``  | 1000000                   |
+-------------------------+---------------------------+

After filling in these variables, your enrichment facility prototype will look like:

.. code-block:: XML

  <facility>
    <name>EnrichmentPlant</name>
    <config>
      <Enrichment>
        <feed_commod>u-ore</feed_commod>
        <feed_recipe>nat-u</feed_recipe>
        <product_commod>fresh-uox</product_commod>
        <tails_commod>tails</tails_commod>
        <max_feed_inventory>1000000</max_feed_inventory>
      </Enrichment>
    </config>
  </facility>

Once complete, append this facility under the Source prototype of your input file.

> Concept: Reactor Prototype

Reviewer: Example?

Concept: Commodities
--------------------

|Cyclus| exchanges resources between facilities using a market-like mechanism
called the **dynamic resource exchange (DRE)**.  The concept of a commodity is
uses to simply indicate which facilities may be interested in trading with
each other through the DRE.  **A commodity is therefore nothing more than a
unique name that is used to define a set of producers and consumers of a
common resource**.  A commodity does not necessarily have a specific
composition; this will be determined by the agents during the simulation.

> Suppliers then respond to the series of requests with a bid . A bid

Reviewer 2: ```suggestion
Suppliers then respond to the series of requests with a bid. A bid
```

which it is congenial; similarly, a region may negate any transfers of
material which have a higher uranium enrichment than is allowable.

For example, the flow graph below shows three suppliers (left) and two
requesters (right), and the potential flows of various commodities among
them. The second consumer makes two different requests. Meanwhile, the
second supplier can supply the commodities requested by both consumers
and provides two bids accordingly.

.. image:: trade.png
    :align: center
    :alt: Commodity trade flowchart

Activity: Create fuel commodities (optional)
+++++++++++++++++++++++++++++++++++++++++++++++++++++

Let's build ``u-ore``, ``fresh-uox``, ``spent-uox``, and ``tails``,
four commodities that will be traded in our simulation. Note that
this part is **optional**, and is only needed if the user wants
to specify the solution priority of each commodity.

* ``u-ore`` : natural uranium that is mined
* ``tails`` : waste from the enrichment process
* ``fresh-uox``: fresh 4.0% enriched Uranium Oxide fuel that enters the reactor
* ``spent-uox``: spent Uranium Oxide fuel that leaves the reactor after depletion

> Whenever |CYCLUS| needs

Reviewer 2: ```suggestion
When |CYCLUS| needs
```

to know the isotopic composition of a material, it looks at the recipe for that
material given in the input file. Until now, ``recipe`` has been used to
refer to fuel recipes, but the ``recipe`` section of the input file can

> include the recipe for natural uranium, spent fuel, fresh fuel, or any

Reviewer 2: ```suggestion
describe any isotopic vector (e.g. natural uranium, spent fuel, fresh fuel, or any
```

> other material where the isotopic composition needs to be tracked.

Reviewer 2: ```suggestion
other material whose isotopic composition needs to be tracked.)
```

> The commodities section is located right under the ``archetype`` section

Reviewer 2: ```suggestion
The commodities section is located below the ``archetype`` section
```

      <commodity>
        <name>com4</name>
        <solution_priority>1.0</solution_priority>
      </commodity>

where:

* name: name of the commodity
* solution priority: number defining the relative priority for resolution in the dynamic
  resource exchange.

 We will model four commodities: u-ore, fresh-uox, spent-uox, and tails.

Activity: Building commodities
++++++++++++++++++++++++++++++++++++++++++

Using the table below and the commodities template above, fill out the commodities
template.

+-------------+-------------+---------------------+
| Commodity   | Name        | Solution Priority   |
+=============+=============+=====================+
| com1        | u-ore       | 1.0                 |
+-------------+-------------+---------------------+
| com2        | fresh-uox   | 1.0                 |
+-------------+-------------+---------------------+
| com3        | spent-uox   | 1.0                 |
+-------------+-------------+---------------------+
| com4        | tails       | 1.0                 |
+-------------+-------------+---------------------+

1. Let's start with ``u-ore``. In the ``<name>`` line replace ``com1`` with ``u-ore``.

.. code-block:: XML

> 

Reviewer 2: ```suggestion
```

    <commodity>
        <name>u-ore</name>

2. In the ``<solution_priority>`` section replace ``val1`` with ``1.0``.

.. code-block:: XML

> 

Reviewer 2: ```suggestion
```

    <commodity>
        <name>u-ore</name>
        <solution_priority>1.0</solution_priority>

3. Now, finalize this commodity by closing it with ``</commodity>``. Your ``u-ore`` commodity section should be:

.. code-block:: XML

> 

Reviewer 2: ```suggestion
```

    <commodity>
        <name>u-ore</name>
        <solution_priority>1.0</solution_priority>
      </commodity>

4. Repeat this process for the other three commodities. Your final result should look like:

.. code-block:: XML

  <commodity>
      <name>u-ore</name>
      <solution_priority>1.0</solution_priority>
    </commodity>
    <commodity>
      <name>fresh-uox</name>
      <solution_priority>1.0</solution_priority>
    </commodity>
    <commodity>
      <name>tails</name>
      <solution_priority>1.0</solution_priority>
    </commodity>
    <commodity>
      <name>spent-uox</name>
      <solution_priority>1.0</solution_priority>
  </commodity>

> Once complete append the commodities section under the archetypes section.

Reviewer 2: ```suggestion
Once complete, append the commodities section under the archetypes section.
```

Concept: Recipes
----------------

Most commodities are materials, which have a quantity and an
isotopic composition.
Recipes are the isotopic composition of a certain material. For

> example, u-ore has an isotropic composition of 0.711% uranium-235 and

Reviewer 2: ```suggestion
example, u-ore has an isotropic composition of 0.711% :math:`^{235}`\ U and
```

> 99.284% uranium-238. The recipe section of a CYCLUS input file is

Reviewer 2: ```suggestion
99.284% :math:`^{238}`\ U. The recipe section of a CYCLUS input file is
```

located at the bottom and is of the form:

.. code-block:: XML

     <recipe>
      <name>nat-u</name>
      <basis>mass</basis>
      <nuclide>
        <id>92235</id>
        <comp>0.00711</comp>
      </nuclide>
      <nuclide>
        <id>92238</id>
        <comp>0.99289</comp>
      </nuclide>
     </recipe>

where ``id`` is the Nuc Id of the isotope in form ZZAAA and ``comp`` is the
composition of that isotope in the recipe. Other isotope formats are

> also acceptable. For example, U-235 can be expressed as:

Reviewer 2: ```suggestion
also acceptable. For example, :math:`^{235}`\ U can be expressed as:
```

* 922350000 (ZZAAAMMMM)
* 92235 (ZZAAA)
* U235 (name)
* U-235 (name)

For more details, reference the `Recipe definition
<../input_specs/recipe.html>`_ page.

First, we can declare the isotopic compositions of the fresh and spent

> fuel. We'll be using simple recipes: fresh fuel is 4.0% U-235 by mass,

Reviewer 2: ```suggestion
fuel. We'll be using simple recipes: fresh fuel is 4.0% :math:`^{235}`\ U by mass,
```

> remainder U-238. Spent fuel is 1.1% U-235, 94.0% U-238, 0.9% Pu-239, and

Reviewer 2: ```suggestion
remainder U-238. Spent fuel is 1.1% :math:`^{235}`\ U, 94.0% :math:`^{238}`\ U, 0.9% :math:`^{239}`\ Pu, and
```

> 4.0% Cs-137.

Reviewer 2: ```suggestion
4.0% :math:`^{137}`\ Cs.
```
